fix: #816 storyMatch traversing directories correctly

waldemarennsaed opened the pull request:
<!-- Thank you for contributing! -->

### Description

<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->

Adjusts the `/packages/histoire/src/node/stories.ts` file to fix traversing through directories to find matching stories for provided `storyMatch` paths.

`storyMatch` works again with relative paths like `storyMatch: ['src/**/*.story.vue']`.
Broken since Vite 6 upgrade in 40c16f2.

The implemented change was checking the currently traversed path for being relative or not. So relative paths are resolved correctly again.

Fixes #816 

How to test:

1. go to `/examples/vue3/histoire.config.ts`
2. add the config: `storyMatch: ['src/**/*.story.vue']`
3. notice how the stories are all there and resolved correctly
4. checkout the current main branch (before this fix) and test the same way -> most of the stories are not resolved

### Additional context

<!-- e.g. is there anything you'd like reviewers to focus on? -->

As there were no tests for `/packages/histoire/src/node/stories.ts`, I did not create a test-file, since _chokidar_ would require actual file-system interaction which goes beyond unit-testing.

### What is the purpose of this pull request? <!-- (put an "X" next to an item) -->

- [x] Bug fix
- [~] New Feature
- [~] Documentation update
- [~] Other

### Before submitting the PR, please make sure you do the following

- [~] If it's a new feature, provide a convincing reason to add it. *Ideally, you should open a suggestion issue first and have it approved before working on it.*
- [x] Read the [Contributing Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [Commit Convention](https://github.com/Akryum/histoire/blob/main/.github/commit-convention.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [~] Ideally, include relevant tests that fail without this PR but pass with it.

waldemarennsaed commented:
One issue I didn't notice, on it.

hugoattal commented:
I admit, I'm not quite sure why it's fixing this 🤔... Isn't this equivalent to `return false;` ?

I can read this from the chokidar github (https://github.com/paulmillr/chokidar?tab=readme-ov-file#path-filtering)
> `ignored function`, regex, or path. Defines files/paths to be ignored. The whole relative or absolute path is tested, not just filename. If a function with two arguments is provided, it gets called twice per path - once with a single argument (the path), second time with two arguments (the path and the [`fs.Stats`](https://nodejs.org/api/fs.html#fs_class_fs_stats) object of that path).

waldemarennsaed commented:
@hugoattal I added a check with `{ relative }` from the `pathe` package which is already being used. It provides a function to easily check for a path being relative or not, through this method: https://nodejs.org/api/path.html#pathrelativefrom-to

I updated the PR description with instructions on how to test.

I also added a nested component + story to the `examples/vue3` example - if not required, I can remove it again.

hugoattal commented on packages/histoire/src/node/stories.ts:
```diff
@@ -48,14 +48,20 @@ export async function watchStories(newContext: Context) {
 
-      return stats?.isFile()
+      // Allow directories to be traversed, only ignore files that don't match
+      return stats?.isFile() ?? false
```
I still don't understand this 🤔... Is this just for typescript inference?
Isn't something like this a bit more readable? `return !!stats?.isFile()`

hugoattal reviewed (commented):
Seems good to me, thanks a lot for the PR 😄 ! Just need a little information on the `?? false`, I'm not sure what this changes 🤔
